Lib/typing.py

```diff
@@ -299,51 +300,22 @@ def __deepcopy__(self, memo):
         return self
 
 
-class _SpecialForm(_Final, _Immutable, _root=True):
-    """Internal indicator of special typing constructs.
-    See _doc instance attribute for specific docs.
-    """
-
-    __slots__ = ('_name', '_doc')
-
-    def __new__(cls, *args, **kwds):
-        """Constructor.
-
-        This only exists to give a better error message in case
-        someone tries to subclass a special typing object (not a good idea).
-        """
-        if (len(args) == 3 and
-                isinstance(args[0], str) and
-                isinstance(args[1], tuple)):
-            # Close enough.
-            raise TypeError(f"Cannot subclass {cls!r}")
```

Comment:
It would be good to keep a custom error message, but not important.

Reply:
You are right, I tested again `class A(Any): pass`, and got "`__init__() takes 2 positional arguments but 4 were given`" which does not look user friendly.

But the former message was "`Cannot subclass <class 'typing._SpecialForm'>`" which does not look correct, because we subclass an instance of `_SpecialForm`, not `_SpecialForm` itself. It is not possible to fix in `__new__`, but we can raise better error in custom `__mro_entries__`.

Reply:
Using `__mro_entries__` is a great idea!